api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -178,3 +182,38 @@ message RoutesConfigDump {
   // The dynamically loaded route configs.
   repeated DynamicRouteConfig dynamic_route_configs = 3 [(gogoproto.nullable) = false];
 }
+
+// This message describes the control plane configuration that Envoy is currently connected to for
+// various xDS implementations.
+message ControlPlaneConfigDump {
+
+  // This message describes the control plane information per each config source.
+  message ConfigSourceControlPlaneInfo {
+
+    // gRPC service represented by ApiConfigSource.
+    envoy.api.v2.core.GrpcService grpc_service = 1;
```

Comment:
I'm not sure if this provides unique identification. If you have lets say two clusters X and Y which uses EDS. IIUC you then have one `ServiceControlPlaneInfo` for EDS and then two `ConfigSourceControlPlaneInfo` inside that. They may point at the same EDS gRPC service but might have separate connections and last updates.

The way things work together, in the above example, they are muxed, but this is really an implementation detail. I'd like to support the idea that you could split different EDS subsets to the same control plane over multiple connections if non-aggregated.

What's missing is the notion of the CDS resource that depends on the EDS object here. This is pretty hard to capture generically, maybe we could have a field that is `parent` that names the resource responsible for creating the xDS request. For CDS/LDS, this might be empty, for SDS/LDS/RDS, this could reflect the `Listener` or `Cluster` responsible.

WDYT? @fredlas also for thinking on this.

Reply:
I don't quite understand. If each cluster gets its own `ConfigSourceControlPlaneInfo`, even if duplicate, doesn't that already support the goal of being agnostic to whether they're being aggregated over the same gRPC channel or even stream?

As for the dependencies... actually, I think I don't entirely understand most of what's going on here. I thought the idea was to just report "whom" Envoy's xDS connections are going to. I would expect that to be either just a bunch of ControlPlane strings, or a bunch of pairs of (ControlPlane string, type URL string).

Reply:
@htuch
>>  IIUC you then have one ServiceControlPlaneInfo for EDS and then two ConfigSourceControlPlaneInfo inside that. They may point at the same EDS gRPC service but might have separate connections and last updates.

In that case I was thinking that there will be only be one `ConfigSourceControlPlaneInfo` inside `ServiceControlPlaneInfo` - That is how the implementation was. It checks if there is a `ConfigSourceControlPlaneInfo` for a gRPC service - it just updates the existing instead of creating a new one. 

Are you thinking we should represent them as two entries?

Reply:
I guess it depends on what you want to do with the information. Do you want to just know who you are connected to coarsely, or do you want to know details on how configuration is being delivered for a specific cluster or listener?

Reply:
@htuch  My main intent was to  to just know who you are connected to coarsely - to which control Envoy is connected.

Reply:
If it's only "I'm connected to this set of servers for these gRPC services", then what you have is fine. Thanks for clarifying.